src/components/pages/career-details/new-career-form/career-form.tsx

```diff
@@ -0,0 +1,324 @@
+import axios, { AxiosResponse } from 'axios';
+import React, { useEffect } from 'react';
+import { ErrorCode, FileError } from 'react-dropzone';
+import { useForm } from 'react-hook-form';
+import { Trans, useTranslation } from 'react-i18next';
+import styled from 'styled-components';
+import { Checkbox } from '../../../forms/checkbox/checkbox';
+import { SIMPLE_EMAIL_PATTERN } from '../../../forms/constants';
+import {
+  FileDropper,
+  FileDropperType,
+} from '../../../forms/file-dropper/file-dropper';
+import { TextArea } from '../../../forms/text-area/text-area';
+import {
+  StyledErrorMessage,
+  TextInput,
+} from '../../../forms/text-input/text-input';
+import { TextStyles } from '../../../typography';
+import { Button } from '../../../ui/buttons/button';
+import { StyledLink } from '../../contact/contact-form';
+import { FormLayout } from '../../contact/form';
+import { SectionHeadline } from '../job-description';
+import { Success } from './career-form-success';
+
+interface CareerFormProps {
+  company_id: string;
+  access_token: string;
+  job_position_id: string;
+  recruiting_channel_id: string;
+  scrollToStart: () => void;
+}
+
+export type FormDataProps = {
+  first_name: string;
+  last_name: string;
+  email: string;
+  documents: FileDropperType[];
+  message: string;
+  phone?: string;
+  location: string;
+  available_from: string;
+  salary_expectations: string;
+
+  privacy: boolean;
+};
+
+type FormErrors = { api?: never };
+
+interface PersonioApiResponse {
+  success: string;
+}
+
+const API_ENDPOINT = 'https://api.personio.de/recruiting/applicant';
+const PRIVACY_POLICY = 'https://satellytes.jobs.personio.de/privacy-policy';
+const MAX_SIZE = 20 * 1024 * 1024;
+
+const TextWrapper = styled.div`
+  margin: 24px 0;
+
+  ${TextStyles.textR}
+`;
+
+export const Form = (props: CareerFormProps) => {
```

Comment:
This will is still a monster. Can you think of ways to divide and conquer? Like putting the request with axios into a distinct file. Or creating smaller components like `CareerFileUpload` that just wraps the following:

```
<FileDropper
          register={register}
          setValue={setValue}
          setError={setError}
          clearErrors={clearErrors}
          errors={errors}
          name="documents"
          fileCategories={fileCategories}
          acceptedFileTypes={'.pdf'}
          illustration="monitor_024"
          maxFiles={3}
          validator={validator}
        />
```

That way you could have a very semantic form described while you push away the details of the implementation into a separate component that you could even collect in a separate file like `form-components`. That's an idea. Maybe it doesn't make sense in the end but we have to do something about the size because I don't feel comfortable looking at it and I don't think people feel comfortable touching and extending it in the future.

```
<!-- other fields -->
<ApplicationFileUploader/>

```

Reply:
I saw the new field `CareerDetailsFileUpload` 🙌